src/Sulu/Bundle/MediaBundle/Resources/js/containers/Form/fields/ImageMap.js

```diff
@@ -0,0 +1,154 @@
+// @flow
+import React from 'react';
+import {observable, toJS} from 'mobx';
+import debounce from 'debounce';
+import userStore from 'sulu-admin-bundle/stores/userStore';
+import type {FieldTypeProps} from 'sulu-admin-bundle/types';
+import type {BlockError} from 'sulu-admin-bundle/containers/Form/types';
+import FieldRenderer from 'sulu-admin-bundle/containers/FieldBlocks/FieldRenderer';
+import jsonpointer from 'json-pointer';
+import ImageMapContainer from '../../ImageMap';
+import type {Value as ImageMapValue, RenderHotspotFormCallback} from '../../ImageMap/types';
+
+const MISSING_TYPE_ERROR_MESSAGE = 'The "image_map" field type needs at least one type to be configured!';
+
+export default class ImageMap extends React.Component<FieldTypeProps<ImageMapValue>> {
+    handleFinish = debounce(() => {
+        const {onFinish} = this.props;
+
+        onFinish();
+    }, 1000);
+
+    handleChange = (value: ImageMapValue) => {
+        const {onChange} = this.props;
+
+        onChange(value);
+        this.handleFinish();
+    };
+
+    get defaultValue(): ImageMapValue {
+        return {
+            imageId: undefined,
+            hotspots: [],
+        };
+    }
+
+    getHotspotFormSchemaType = (type: ?string) => {
+        const {defaultType, schemaPath, types} = this.props;
+
+        if (!type) {
+            throw new Error(
+                'It is impossible that a hotspot has no formType. This should not happen and is likely a bug.'
+            );
+        }
+
+        if (!types) {
+            throw new Error(MISSING_TYPE_ERROR_MESSAGE);
+        }
+
+        if (types[type]) {
+            return types[type];
+        }
+
+        if (!defaultType) {
+            throw new Error(
+                'It is impossible that a image_map has no defaultType. This should not happen and is likely a bug.'
+            );
+        }
+
+        if (!types[defaultType]) {
+            throw new Error(
+                'The default type should exist in image_map "' + schemaPath + '".'
+            );
+        }
+
+        return types[defaultType];
+    };
+
+    handleHotspotFormChange = (index: number, name: string, value: Object) => {
+        const {onChange, value: oldValues} = this.props;
+
+        if (!oldValues) {
+            return;
```

Comment:
This can't happen since we call `onChange` with the default value immediately, if value is undefined

Reply:
This is just for flow i guess

Reply:
Would still throw an error then... Because in case we have a bug and it would execute that return, nothing would happen although it should. Spotting the error would be much harder if we just return here, because there would not even be an error shown anywhere.